src/slicer.cpp

```diff
@@ -82,122 +84,305 @@ int SlicerLayer::getNextSegmentIdx(const Mesh* mesh, const SlicerSegment& segmen
 
 void SlicerLayer::connectOpenPolylines(Polygons& open_polylines)
 {
-    // TODO use some space partitioning data structure to make this run faster than O(n^2)
-    for(unsigned int open_polyline_idx = 0; open_polyline_idx < open_polylines.size(); open_polyline_idx++)
+    bool allow_reverse = false;
+    // Search a bit fewer cells but at cost of covering more area.
+    // Since acceptance area is small to start with, the extra is unlikely to hurt much.
+    coord_t cell_size = largest_neglected_gap_first_phase * 2;
+    connectOpenPolylinesImpl(open_polylines, largest_neglected_gap_second_phase, cell_size, allow_reverse);
+}
+
+void SlicerLayer::stitch(Polygons& open_polylines)
+{
+    bool allow_reverse = true;
+    connectOpenPolylinesImpl(open_polylines, max_stitch1, max_stitch1, allow_reverse);
+}
+
+void SlicerLayer::connectOpenPolylinesImpl(Polygons& open_polylines, coord_t max_dist, coord_t cell_size, bool allow_reverse)
```

Comment:
This function is about 300 lines. Could you maybe try to cut it up into smaller pieces?

Breaking the code up into bit size chunks and adding some documentation might help me a lot in understanding the code.